ui/components/NodeInputText.tsx

```diff
@@ -33,17 +42,18 @@ export const NodeInputText: FC<NodeInputProps> = ({
       name={attributes.name}
       label={getNodeLabel(node)}
       disabled={disabled}
-      defaultValue={node.messages.map(({ text }) => text).join(" ")}
+      defaultValue={defaultValue}
       error={
-        attributes.name === "code" ||
+        messageError ??
+        (attributes.name === "code" ||
         attributes.name === "totp" ||
         attributes.name === "totp_code" ||
         attributes.name === "lookup_secret" ||
         (isWebauthn && attributes.name === "identifier")
           ? ucFirst(error)
           : isDuplicate
             ? "This value is already in use"
-            : undefined
+            : undefined)
```

Comment:
issue: this is pretty hard to decipher, can we move this behavior in a separate function?

Reply:
Very good idea. I extracted this to make it more readable.